File: 2025/50. Pow(x, n)/Solution2.java
class Solution {
    public double myPow(double x, int n) {
        if (n < 0) {
            n = -n;
            x = 1 / x;
        }
        return helper(x, (long) n);
    }

    private double helper(double x, long n) {
        if (x == 0)
            return 0.0;
        if (n == 0)
            return 1.0;
        if (n % 2 == 1)
            return x * helper(x * x, (n - 1) / 2);
        else
            return helper(x * x, n / 2);
    }
}

File: 2025/1770. Maximum Score from Performing Multiplication Operations/solution2.py
class Solution:
    def maximumScore(self, nums: List[int], multipliers: List[int]) -> int:
        
        n, m = len(nums), len(multipliers)

        @lru_cache(maxsize=None)
        def helper(left, right, ops):
            if ops == m:
                return 0
            mt = multipliers[ops]
            return max(
                nums[left] * mt + helper(left + 1, right, ops + 1),
                nums[right] * mt + helper(left, right - 1, ops + 1)
            )

        return helper(0, n - 1, 0)